tests/TestCase/ORM/Association/BelongsToManyTest.php

```diff
@@ -828,6 +828,69 @@ public function testReplaceLinkSuccess()
         $this->assertFalse($entity->dirty('tags'));
     }
 
+
+    /**
+     * Tests that replaceLinks() will contain() the target table when
+     * there are conditions present on the association.
+     *
+     * @return void
+     */
+    public function testReplaceLinkWithConditions()
+    {
+        $connection = ConnectionManager::get('test');
+        $joint = $this->getMock(
+            '\Cake\ORM\Table',
+            ['find'],
+            [['alias' => 'ArticlesTags', 'connection' => $connection]]
+        );
+        $config = [
+            'sourceTable' => $this->article,
+            'targetTable' => $this->tag,
+            'through' => $joint,
+            'joinTable' => 'tags_articles',
+            'conditions' => ['Tags.id' => 'blah'],
+        ];
+        $assoc = $this->getMock(
+            '\Cake\ORM\Association\BelongsToMany',
+            ['_collectJointEntities'],
+            ['tags', $config]
+        );
+        $assoc->junction();
+
+        $query1 = $this->getMock(
+            '\Cake\ORM\Query',
+            ['where', 'andWhere', 'addDefaultTypes', 'contain'],
+            [$connection, $joint]
+        );
+        $query1->expects($this->at(0))
+            ->method('where')
+            ->will($this->returnSelf());
+        $query1->expects($this->at(1))
+            ->method('where')
+            ->will($this->returnSelf());
+        $query1->expects($this->at(2))
+            ->method('contain')
+            ->with('Tags')
+            ->will($this->returnSelf());
+        $query1->expects($this->at(3))
+            ->method('andWhere')
+            ->with($config['conditions'])
+            ->will($this->returnSelf());
+        $query1->setResult(new \ArrayIterator([]));
+
+        $joint->expects($this->at(0))->method('find')
+            ->with('all')
+            ->will($this->returnValue($query1));
+
+        $entity = new Entity(['id' => 1, 'test' => []]);
+
+        $assoc->expects($this->once())->method('_collectJointEntities')
+            ->with($entity, [])
+            ->will($this->returnValue([]));
+
+        $assoc->replaceLinks($entity, [], ['associated' => false]);
```

Comment:
Well it does, but with a kind of strange description: "Mocked method does not exist."

```
$ vendor/bin/phpunit tests/TestCase/ORM/Association/BelongsToManyTest.php
 --filter testReplaceLinkWith
PHPUnit 5.0.8 by Sebastian Bergmann and contributors.

F                                                                   1 / 1 (100%)

Time: 854 ms, Memory: 9.50Mb

There was 1 failure:

1) Cake\Test\TestCase\ORM\Association\BelongsToManyTest::testReplaceLinkWithConditions
Expectation failed for method name is equal to <string:contain> when invoked at sequence 
index 2.
Mocked method does not exist.

FAILURES!
Tests: 1, Assertions: 3, Failures: 1.
```

Admittedly, I'd be happier with a test that actually attempted to run the resulting SQL that included the target table's alias to ensure it was executable, but that's probably not appropriate or practical here.